### PR TITLE
[Feature] Create/Use placement groups for deploys.

### PR DESCRIPTION
**Pull request by grahamjenson**
This feature will create/use placement groups for AutoScalingGroups. This PR: 
1. Creates the placement group if not yet created (if created validates the prefix is `odin/<project_name>/<config_name>`)
2.  If already existing checks the values provided are the same to make sure we know the values. (we don't do any kind of updating because might cause bad situations where two projects update the placemnet group causing lots of termed instances) 

There are many caveats to using placement groups when deploying, read about them https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/placement-groups.html#placement-groups-cluster

### CODE REVIEW

File: aws/pg/placement_group.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	pg, err := findPlacementGroup(ec2c, groupName)
+	if err != nil {
+		return err
+	}
+
+	// If no pg is found create a new one
+	// otherwise validate that the found one equals the correct values
+	if pg == nil {
+		// If we are creating the Placement Group we restrict the name to prefix
+		if !strings.HasPrefix(*groupName, prefix) {
```
Comment (chwevans) [inline aws/pg/placement_group.go]: Would be nice if validation was outside here

File: aws/pg/placement_group.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			return fmt.Errorf("PlacementGroupError(%s): If this is a new PlacementGroup it must have %q prefix", to.Strs(groupName), prefix)
+		}
+
+		return createNewPlacementGroup(ec2c, groupName, partitionCount, strategy)
```
Comment (chwevans) [inline aws/pg/placement_group.go]: Validate checks that it is available at the end of this function? Will this one be available in time?

Comment (grahamjenson) [reply aws/pg/placement_group.go]: yes, placement groups are not much more than a name, so creation is basically instant

File: aws/pg/placement_group.go (hunk 1 above)
Comment (grahamjenson) [reply aws/pg/placement_group.go]: We want to allow projects to use other projects placement groups, I just want to ensure that placement groups created by odin have the correct prefix. 

This is not ideal, but doing any sort of auto-name assignment makes this harder to do.

**Pull request merged**